src/pages/collators/Collators.tsx

```diff
@@ -47,12 +48,13 @@ export function Collators() {
   const { candidates, inflationInfo, estimatedRewards } = useStakingPallet();
 
   // Holds the candidate for which the delegation modal is to be shown
-  const [selectedCandidateForDelegation, setSelectedCandidateForDelegation] =
+  const [selectedCandidate, setSelectedCandidate] =
     useState<ParachainStakingCandidate | undefined>(undefined);
 
   const [userAvailableBalance, setUserAvailableBalance] = useState<string>("0.00");
   const [userStaking, setUserStaking] = useState<UserStaking>();
   const [claimDialogOpen, setClaimDialogOpen] = useState<boolean>(false);
```

Comment:
Can we maybe add a comment here that this is basically a switch between bonding/unbonding mode? I get how and that it works but it's a bit implicit and in a few weeks getting back to this might be confusing. 
We could also change it to something like this
```
  const [bondingMode, setBondingMode] = useState<'bonding' | 'unbonding' | undefined>(undefined);
```

Reply:
It is true that there's room for improvement here.
The way I see it, _selectedCandidate_ is selected in any case you interact with a candidate in a row. It's just the 'selected row' sort of, no matter what you do with that info later.

I am using isDelegatingMore and isDelegatingLess to manage the state internally in the dialogs. In that case, unbonding is implied by isDelegatingLess, and the other is implied by the negation of isDelegatingLess. It's its not perfect, I even considered replacing the booleans with a state like the one you suggest, that contemplates three states: bonding, bondingMore, unbonding, but I thought it was not worth it. Let me know your thoughs @ebma